src/CoreShop2VueStorefrontBundle/Bridge/DocumentMapper/DocumentCategoryMapper.php

```diff
@@ -42,36 +40,30 @@ public function __construct(
 
     /**
      * @param \CoreShop\Component\Product\Model\CategoryInterface $category
-     * @param int $level
-     * @param int $position
-     * @param int $parentId
-     *
-     * @return Category
      */
-    public function mapToDocument(
-        $category,
-        $level = self::CATEGORY_DEFAULT_LEVEL,
-        $position = self::CATEGORY_DEFAULT_POSITION,
-        $parentId = self::CATEGORY_DEFAULT_PARENT_ID
-    ): Category {
+    public function mapToDocument($category): Category
+    {
         $esCategory = $this->documentFactory->getOrCreate(Category::class, $category->getId());
 
         $categoryName = $category->getName() ?: $category->getKey();
+        $parentCategory = $category->getParentCategory();
+
+        $level = substr_count($category->getFullPath(), '/') - 1;
 
         $esCategory->setId($category->getId());
-        $esCategory->setParentId($parentId);
+        $esCategory->setParentId($parentCategory ? $parentCategory->getId() : 0);
         $esCategory->setName($categoryName);
         $esCategory->setLevel($level);
         $esCategory->setCreatedAt($this->getDateTime($category->getCreationDate()));
         $esCategory->setUpdatedAt($this->getDateTime($category->getModificationDate()));
         $esCategory->setPath($this->documentHelper->buildPath($category));
         $esCategory->setDisplayMode(self::CATEGORY_DEFAULT_DISPLAY_MODE);
         $esCategory->setPageLayout(self::CATEGORY_DEFAULT_PAGE_LAYOUT);
-        $esCategory->setChildrenData($this->buildChildrenData($category->getChildCategories(), ++$level));
+        $esCategory->setChildrenData($this->buildChildrenData($category->getChildCategories(), $level + 1));
         $esCategory->setChildren($this->documentHelper->buildChildrenIds($category->getChildCategories()));
         $esCategory->setChildrenCount($this->documentHelper->buildChildrenCount($esCategory->children));
-        $esCategory->setIsAnchor("0");
-        $esCategory->setPosition($position);
+        $esCategory->setIsAnchor("1");
+        $esCategory->setPosition($category->getIndex());
```

Comment:
`getIndex` is available on Pimcore objects if you enable sorting manually on the parent.